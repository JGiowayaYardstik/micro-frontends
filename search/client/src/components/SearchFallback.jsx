import React, { Component } from 'react';
import PropTypes from 'prop-types';
import SearchFallbackSuggestions from './SearchFallbackSuggestions';
import isEmpty from '../util/isEmpty';
import getObjectValues from '../util/getObjectValues';

class SearchFallback extends Component {
  getSuggestions = () => {
    const { searchState } = this.props;
    const suggestions = [];
    if (!isEmpty(searchState)) {
      getObjectValues(searchState).forEach(entry => {
        if (entry.suggestions && entry.suggestions.length > 0) {
          entry.suggestions.forEach(suggestion => suggestions.push(suggestion));
        }
      });
    }
    return suggestions;
  };

  render() {
    const suggestions = this.getSuggestions();

    const query = decodeURIComponent(this.props.query);

    return (
      <div>
        <h1>{`Keine Treffer für "${query}"`}</h1>
        {suggestions && suggestions.length > 0 && (
          <SearchFallbackSuggestions suggestions={suggestions} />
        )}
        <p>
          <h2>Suchtipps</h2>
          <div>
            Versuchen Sie, allgemeiner zu suchen - Sie können anschließend die Suchergebnisse
            filtern.
          </div>
        </p>
      </div>
    );
  }
}

SearchFallback.propTypes = {
  query: PropTypes.string,
  searchState: PropTypes.object,
};

export default SearchFallback;
