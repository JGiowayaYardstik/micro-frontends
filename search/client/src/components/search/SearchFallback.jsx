import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withI18n } from 'react-i18next';
import { connect } from '@dm/style-provider';
import { compose } from 'recompose';
import { getContentProvider } from 'search-api';
import SearchFallbackSuggestions from './SearchFallbackSuggestions';
import isEmpty from '../../util/isEmpty';
import getObjectValues from '../../util/getObjectValues';

class SearchFallback extends Component {
	componentDidUpdate() {
		this.props.handleUpdateSuggestions();
	}

	getSuggestions = () => {
		const { searchState } = this.props;
		const suggestions = [];
		if (!isEmpty(searchState)) {
			getObjectValues(searchState).forEach(entry => {
				if (entry.suggestions && entry.suggestions.length > 0) {
					entry.suggestions.forEach(suggestion => suggestions.push(suggestion));
				}
			});
		}
		return suggestions;
	};

	renderSearchFallback = () => {
		const { searchFallback, basicConfig, styles } = this.props;

		if (!searchFallback) {
			return null;
		}

		return getObjectValues(searchFallback).map(contentId => (
			<div
				data-dmid="search-fallback-info-container-content"
				className={styles.infoContainerContent}
				key={contentId}>
				{getContentProvider() && getContentProvider().getContentComponent(contentId, basicConfig)}
			</div>
		));
	};

	render() {
		const { styles, t } = this.props;
		const suggestions = this.getSuggestions();

		const query = decodeURIComponent(this.props.query);

		return (
			<div data-dmid="search-fallback-container" className={styles.searchFallbackContainer}>
				<h1 data-dmid="search-fallback-headline" className={styles.headline}>
					{t('search.fallback.headline', { query })}
				</h1>
				{suggestions &&
					suggestions.length > 0 && <SearchFallbackSuggestions suggestions={suggestions} />}
				<div data-dmid="search-fallback-info-container" className={styles.infoContainer}>
					{this.renderSearchFallback()}
				</div>
			</div>
		);
	}
}

const searchFallbackContainer = ({ theme }) => ({
	column2to6: {
		padding: `0 ${theme.dimension.spacing.xs.rem}`
	}
});

const headline = ({ theme }) => ({
	hyphens: 'auto',
	fontFamily: theme.typography.fontFamily.primary.medium,
	fontSize: theme.typography.fontSize.xxl.rem,
	color: theme.palette.color2.hex,
	textAlign: 'center',
	margin: `${theme.dimension.spacing.m.rem} 0 ${theme.dimension.spacing.xl.rem}`
});

const infoContainer = () => ({
	display: 'flex',
	flexWrap: 'wrap',
	justifyContent: 'space-around',
	position: 'relative',
	marginTop: '0'
});

const infoContainerContent = () => ({
	flex: '0 0 auto',
	width: '30%',

	'&:first-of-type': {
		paddingRight: '2rem'
	},

	column2to6: {
		flex: '1 0 100%',

		'&:first-of-type': {
			paddingRight: 0
		}
	}
});

SearchFallback.propTypes = {
	query: PropTypes.string,
	searchState: PropTypes.object,
	searchFallback: PropTypes.object,
	basicConfig: PropTypes.object,
	handleUpdateSuggestions: PropTypes.func.isRequired,
	styles: PropTypes.object,
	t: PropTypes.func
};

export default compose(
	connect({ headline, infoContainer, infoContainerContent, searchFallbackContainer }),
	withI18n()
)(SearchFallback);
