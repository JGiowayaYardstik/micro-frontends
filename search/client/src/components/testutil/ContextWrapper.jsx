import React from 'react';
import PropTypes from 'prop-types';
import { StyleProvider } from '@dm/style-provider';
import { DeviceProvider } from '@dm/device-provider';
import { I18nextProvider } from 'react-i18next';
import { Provider } from 'react-redux';
import { theme as dm } from '@dm/ui-theme';

export default function ContextWrapper({ children, locale }) {
	return (
		<StyleProvider>
			<DeviceProvider>
				<I18nextProvider i18n={global.i18next} initialLanguage={locale}>
					{children}
				</I18nextProvider>
			</DeviceProvider>
		</StyleProvider>
	);
}

ContextWrapper.propTypes = {
	children: PropTypes.node,
	locale: PropTypes.string
};

export function ContextWrapperWithoutResponsive({ children, locale }) {
	return (
		<StyleProvider>
			<I18nextProvider i18n={global.i18next} initialLanguage={locale}>
				{children}
			</I18nextProvider>
		</StyleProvider>
	);
}

ContextWrapperWithoutResponsive.propTypes = {
	children: PropTypes.node,
	locale: PropTypes.string
};

export function ContextWrapperWithStore({ children, locale, theme = dm, store = global.store }) {
	return (
		<Provider store={store}>
			<StyleProvider theme={theme}>
				<I18nextProvider i18n={global.i18next} initialLanguage={locale}>
					{children}
				</I18nextProvider>
			</StyleProvider>
		</Provider>
	);
}

ContextWrapperWithStore.propTypes = {
	children: PropTypes.node,
	locale: PropTypes.string,
	store: PropTypes.object,
	theme: PropTypes.object
};
