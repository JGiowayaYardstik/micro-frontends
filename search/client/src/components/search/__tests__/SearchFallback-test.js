import React from 'react';
import { createSnapshot } from '@dm/jest-utils';
import ContextWrapper from '../../testutil/ContextWrapper';
import SearchFallback from '../SearchFallback';
import initializeTestContentProvider from '../../testutil/searchContainer/TestContentProviderInitializer';

describe('SearchFallback', () => {
  beforeAll(() => {
    initializeTestContentProvider();
  });

  it('renders and matches snapshot (DE)', () => {
    const snapshot = createSnapshot(
      <ContextWrapper locale="de">
        <SearchFallback
          query="2ß9358ß239648"
          searchFallback={{
            textId1: 535460,
            textId2: 535470,
          }}
          searchState={{ PRODUCT: { count: 0, suggestions: ['hello', 'world'], results: [] } }}
          handleUpdateSuggestions={() => {}}
        />
      </ContextWrapper>
    );
    expect(snapshot).toMatchSnapshot();
  });

  it('renders and matches snapshot (HU)', () => {
    const snapshot = createSnapshot(
      <ContextWrapper locale="hu">
        <SearchFallback
          query="2ß9358ß239648"
          searchFallback={{
            textId1: 535460,
            textId2: 535470,
          }}
          handleUpdateSuggestions={() => {}}
        />
      </ContextWrapper>
    );
    expect(snapshot).toMatchSnapshot();
  });
});
