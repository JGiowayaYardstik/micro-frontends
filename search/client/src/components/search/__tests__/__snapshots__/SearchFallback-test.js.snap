// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFallback renders and matches snapshot (DE) 1`] = `
Object {
  "_locale": "de",
  "_width": 1200,
  "html": "
<div
  data-dmid=search-fallback-container
  className=searchFallbackContainer__1dsx585
>
  <h1 data-dmid=search-fallback-headline className=headline__12u4gl8>
    Ihre Suche nach „2ß9358ß239648” ergab leider keine Treffer
  </h1>
  <div data-dmid=search-fallback-suggestions className=suggestions__e925jf>
    Meinten Sie:{ }
    <a title=hello data-dmid=suggestion-link href=q=hello>
      hello
    </a>,{ }
    <a title=world data-dmid=suggestion-link href=q=world>
      world
    </a>?
  </div>
  <div
    data-dmid=search-fallback-info-container
    className=infoContainer__12ybgy7
  >
    <div
      data-dmid=search-fallback-info-container-content
      className=infoContainerContent__18pnl22
    >
      <div>content component 535460</div>
    </div>
    <div
      data-dmid=search-fallback-info-container-content
      className=infoContainerContent__18pnl22
    >
      <div>content component 535470</div>
    </div>
  </div>
</div>;
",
  "styles": "
.headline__12u4gl8 {
  -webkit-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
  font-family: dmbrand-medium, Arial, Helvetica, sans-serif;
  font-size: 1.75rem;
  color: #002878;
  text-align: center;
  margin: 1.5625rem 0 2.5rem;
}
.infoContainer__12ybgy7 {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-wrap: wrap;
  -webkit-box-lines: multiple;
  justify-content: space-around;
  -webkit-box-pack: justify;
  position: relative;
  margin-top: 0;
}
.infoContainerContent__18pnl22:first-of-type {
  padding-right: 2rem;
}
.infoContainerContent__18pnl22 {
  flex: 0 0 auto;
  width: 30%;
}
.suggestions__e925jf {
  font-size: 1.125rem;
  font-family: dmbrand-medium, Arial, Helvetica, sans-serif;
  margin-bottom: 2.5rem;
  text-align: center;
}
@media (max-width: 959px) {
  .infoContainerContent__18pnl22:first-of-type {
    padding-right: 0;
  }
  .infoContainerContent__18pnl22 {
    flex: 1 0 100%;
  }
  .searchFallbackContainer__1dsx585 {
    padding: 0 0.9375rem;
  }
}
",
}
`;

exports[`SearchFallback renders and matches snapshot (HU) 1`] = `
Object {
  "_locale": "de",
  "_width": 1200,
  "html": "
<div
  data-dmid=search-fallback-container
  className=searchFallbackContainer__1dsx585
>
  <h1 data-dmid=search-fallback-headline className=headline__12u4gl8>
    A(z) „2ß9358ß239648” keresése nem eredményezett találatot.
  </h1>
  <div
    data-dmid=search-fallback-info-container
    className=infoContainer__12ybgy7
  >
    <div
      data-dmid=search-fallback-info-container-content
      className=infoContainerContent__18pnl22
    >
      <div>content component 535460</div>
    </div>
    <div
      data-dmid=search-fallback-info-container-content
      className=infoContainerContent__18pnl22
    >
      <div>content component 535470</div>
    </div>
  </div>
</div>;
",
  "styles": "
.headline__12u4gl8 {
  -webkit-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
  font-family: dmbrand-medium, Arial, Helvetica, sans-serif;
  font-size: 1.75rem;
  color: #002878;
  text-align: center;
  margin: 1.5625rem 0 2.5rem;
}
.infoContainer__12ybgy7 {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-wrap: wrap;
  -webkit-box-lines: multiple;
  justify-content: space-around;
  -webkit-box-pack: justify;
  position: relative;
  margin-top: 0;
}
.infoContainerContent__18pnl22:first-of-type {
  padding-right: 2rem;
}
.infoContainerContent__18pnl22 {
  flex: 0 0 auto;
  width: 30%;
}
@media (max-width: 959px) {
  .infoContainerContent__18pnl22:first-of-type {
    padding-right: 0;
  }
  .infoContainerContent__18pnl22 {
    flex: 1 0 100%;
  }
  .searchFallbackContainer__1dsx585 {
    padding: 0 0.9375rem;
  }
}
",
}
`;
